crates/sui-sdk/examples/transaction_builder_api.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+mod utils;
+use shared_crypto::intent::Intent;
+use sui_config::{sui_config_dir, SUI_KEYSTORE_FILENAME};
+use sui_keys::keystore::{AccountKeystore, FileBasedKeystore};
+use sui_sdk::{
+    rpc_types::SuiTransactionBlockResponseOptions,
+    types::{quorum_driver_types::ExecuteTransactionRequestType, transaction::Transaction},
+    SuiClientBuilder,
+};
+use utils::example_utils::setup_for_write;
+
+// This example shows how to use programmable transaction blocks to chain multiple
+// actions into one transaction. Specifically, the example retrieves two addresses
+// from the local wallet, and then
+// 1) finds a coin from the active address that has Sui,
+// 2) splits the coin into one coin of 1000 MIST and the rest,
+// 3  transfers the split coin to second Sui address,
+// 4) signs the transaction,
+// 5) executes it.
+// For some of these actions it prints some output.
+// Finally, at the end of the program it prints the number of coins for the
+// Sui address that received the coin.
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let sui = SuiClientBuilder::default().build_testnet().await?; // testnet Sui network
+    println!("Sui testnet version {:?}", sui.api_version());
+
+    let (sui, sender, recipient, coin_obj_id) = setup_for_write().await?;
+
+    // Programmable transactions allows the user to bundle a number of actions into one transaction
+    let txb = sui.transaction_builder();
+
+    // 2) Split coin
+    let txb_res = txb
+        .split_coin(sender, coin_obj_id, vec![1000], None, 5_000_000)
+        .await?;
+    println!("Split coin transaction data: {:?}\n\n", txb_res);
+
+    // TODO ge the right object here, which is the new split coin
+    // 3) Transfer object
+
+    let txb_res = txb
+        .transfer_object(sender, coin_obj_id, None, 5_000_000, recipient)
+        .await?;
+    println!("Transfer object data: {:?}\n\n", txb_res);
```

Comment:
I just realised that you have the same issue here as in the `split_coin_digest` util.  The `transaction_builder` API does not expose the API to build a PTB transaction-by-transaction.  It is a legacy API that used to call the `unsafe_` RPC endpoints to create whole PTBs that do single transactions.  To preserve backwards compatibility, it was re-implemented to do the same thing locally, but probably what you want to use is `sui_types::programmable_transaction_builder::ProgrammableTransactionBuilder` -- see my comment on `split_coin_digest` for more details.

Currently this example creates. `split_coin` transaction, discards it, and then creates a `transfer_object` transaction which it actually runs.